plugins/woocommerce/src/StoreApi/Utilities/CheckoutTrait.php

```diff
@@ -220,5 +225,16 @@ private function persist_additional_fields_for_order( \WP_REST_Request $request
 		if ( $errors->has_errors() ) {
 			throw new RouteException( 'woocommerce_rest_checkout_invalid_additional_fields', $errors->get_error_messages(), 400 );
 		}
+
+		// If the user is logged in, we need to sync the customer additional fields with the order
+		// otherwise they will be overwritten on next page load.
+		if ( is_user_logged_in() && 0 !== $this->order->get_customer_id() && get_current_user_id() === $this->order->get_customer_id() ) {
```

Comment:
I don't think you need the first `is_user_logged_in() && ` because the other statements take care of this.

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/53553/commits/405c6813fd1607c2cacff4d08842299abf87c112